hoomd/filter/custom.py

```diff
@@ -0,0 +1,51 @@
+"""Contains a class for custom particle filters in Python."""
+from abc import ABCMeta, abstractmethod
+
+class CustomFilter(metaclass=ABCMeta):
+    """Abstract base class for custom particle filters.
+
+    The class allows the definition of particle filters in Python (see
+    `hoomd.filter.ParticleFilter`.
+    """
+    @abstractmethod
+    def __call__(self, state):
+        """Return the local particle tags that match the filter.
+
+        This can either return the tags that are local to an MPI rank or the
+        tags of all particles that match in the entire state.  Tag numbers in a
+        `hoomd.Snapshot` object are just their index.
+
+        Note:
+            The exact requirements for custom filters is that the set union of
+            the returned arrays from each MPI rank be all particles that match
+            the filter. For general use, it is recommended, however, that each
+            rank only return the tags for particles that are in the local MPI
+            rank (excluding ghost particles).
+
+        Args:
+            state (`hoomd.State`):
+                The simulation state to return the filtered tags from.
+
+        Returns:
+            (*N*,) `numpy.ndarray` of `numpy.uint64`:
```

Comment:
Why the extra asterisks around `N`? That `N` should already be italicized when the docs are formatted (for example, see the [freud cluster docs](https://freud.readthedocs.io/en/stable/modules/cluster.html#freud.cluster.Cluster.cluster_idx), is there another reason I'm missing?